### PR TITLE
azure: Upgrade azidentity library and dependencies to 1.8.0

### PR DESCRIPTION
**Pull request by benbardin**
[Manually ran available Azure roachtests here](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/17710067?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildTestsSection=true).

([One test failed, known flake.](https://teamcity.cockroachdb.com/test/-8975467554695702336?currentProjectId=Cockroach_Nightlies&expandTestHistoryChartSection=true&pager.currentPage=1&orderBy=status&order=desc&expandedTest=build:(id:17710067),id:6136))

[CRDB cloud tests here](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/17783756?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildTestsSection=true) One failure, but is also exists on master

### CODE REVIEW

**Review by adwittumuluri — APPROVED** [review]

**Comment by msbutler** [comment]
@benbardin did you run the cloud unit tests as well?

Also, could you update the commit message to follow protocol? https://cockroachlabs.atlassian.net/wiki/spaces/CRDB/pages/73072807/Git+Commit+Messages

**Comment by benbardin** [comment]
Updated commit message.

Adwit didn't think there was anything reasonable to do beyond the testing I've done - roachtests, unit tests, manual tests. It will be very hard to make the cloud tests work with an unpublished SHA, so as far as cloud goes I don't think there's much we can reasonably do until we cut a 25.1 beta.

File: go.mod [hunk 1]
```diff
@@ -10,14 +10,14 @@ go 1.22.8
 // merge of any upgrades we should communicate to all teams to be on the lookout
 // for behavior changes, just like we would after a go upgrade.
 require (
-	golang.org/x/crypto v0.26.0
+	golang.org/x/crypto v0.27.0
```
Comment (msbutler) [inline go.mod:13]: argh, note the above comment. these core dep upgrades need to be in their own commit. I wonder if we'll get these upgrades for free when dev inf upgrades to go 1.23. Maybe ask dev inf on their timeline for that?

**Review by msbutler — CHANGES_REQUESTED** [review]
blocking on core upgrades in diff

File: go.mod (hunk 1 above)
Comment (benbardin) [reply go.mod:13]: oh, very good eye. thanks for noticing!

**Comment by benbardin** [comment]
We can just use azure 1.7.0, it doesn't need the crypto upgrade and we only need to get >=1.6. Thanks!

https://github.com/cockroachdb/cockroach/pull/135322
